src/clj/salttoday/db/comments.clj

```diff
@@ -0,0 +1,174 @@
+        ; Create a map as such {["comment text" "username"] comment-id}
+        db-text-name-vec-to-id-map (reduce #(assoc %1 [(:comment/text %2) (:user/name %2)] (:db/id %2)) {} db-comments)
+        db-comments-set (set (keys db-text-name-vec-to-id-map)) ; get the keys as a set so we can do a diff with the new comments
+        scraped-comments-set (set (for [comment comments] [(:text comment) (:username comment)])) ; create a set of text and usernames
+        comment-difference (set/difference db-comments-set scraped-comments-set)] ; find which comments are in the db but not scraped
+    (if (pos? (count comment-difference))                   ; if we found comments we didn't scrape, transact them
+      (doseq [text-name-vec comment-difference]
+        (tx-with-logging conn [{:db/id (get db-text-name-vec-to-id-map text-name-vec)
+                                :comment/deleted true}]))))
+  (doseq [comment comments]
+    (add-comment conn post-id post-title comment)))
+
+(def initial-get-all-comments-query
+  "Gets every comment in the database"
+  '{:find [(pull ?c [:db/id :comment/upvotes :comment/downvotes :comment/text :comment/deleted])
+           (pull ?u [:user/name])
+           (pull ?p [:post/title :post/url])]
+    :in [$]
+    :args []
+    :where [[?c :comment/text ?text ?tx]
+            [?c :comment/user ?u]
+            [?p :post/comment ?c]]})
+
+; TODO - probably better if this function took a map of keys instead of positional args
+(defn create-get-comments-query
+  "Adds any optional args/conditionals to the query"
+  [db days-ago-date search-text name cid deleted]
+  (cond-> initial-get-all-comments-query
+    true
+    (update :args conj db)
+
+    (not= 0 cid)
+    (-> (update :in conj '?c)
+        (update :args conj cid))
+
+    search-text
+    (-> (update :in conj '?search-text)
+        (update :args conj search-text)
+        (update :where conj '[(.contains ^String ?text ?search-text)]))
+
+    name
+    (-> (update :in conj '?name)
+        (update :args conj name)
+        (update :where conj '[?u :user/name ?name]))
+
+    deleted
+    (-> (update :where conj '[?c :comment/deleted true]))
+
+    days-ago-date
+    (-> (update :in conj '?days-ago-date)
+        (update :args conj days-ago-date)
+        (update :where conj '[?tx :db/txInstant ?inst])
+        (update :where conj '[(.before ^java.util.Date ?days-ago-date ?inst)]))
+
+    true
+    remap-query))
+
+(defn get-comments
+  "Retrieve comments from the database."
+  ([db days-ago search-text name id deleted]
+   (let [days-ago-date (get-date days-ago)
+         query-map (create-get-comments-query db days-ago-date search-text name id deleted)]
+     (let [results (apply (partial d/q (:query query-map)) (:args query-map))]
+       (db-comments->frontend-comments results))))
+  ([db]
+   (get-comments db -1 nil nil nil false)))
+
+; TODO - we must pass in a keyword for sorting!!!
+(defn get-comments-paginated
```

Comment:
Prefer just get-comments, considering this does more then just paginate

Reply:
above is a function also named `get-comments`

Reply:
Perhaps `get-sorted-paginated-comments`?